wagtail/snippets/models.py

```diff
@@ -106,6 +106,7 @@ def _register_snippet_immediately(model, viewset=None):
     )
 
     viewsets.register(admin_viewset)
+    model.snippet_viewset = admin_viewset
```

Comment:
A bit hesitant to add this, but the problem is that the bulk actions view instance is shared across all the snippet models, and it has its own registration mechanism. Thus, the icon needs to be resolved at request time rather than during the view class's instantiation. Adding this allows us to do `self.model.snippet_viewset.icon` in `get_context_data()` of the view.